bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieClient.java

```diff
@@ -150,6 +150,16 @@ private int getRc(int rc) {
         return faultyBookies;
     }
 
+    public boolean isWritable(BookieSocketAddress address) {
+        final PerChannelBookieClientPool pcbcPool = lookupClient(address);
+        if (pcbcPool == null) {
+            // let write fail with whatever error it produces
+            return true;
+        }
+
+        return pcbcPool.isWritable(0L /*not really used in obtain()*/);
```

Comment:
If we have multiple channels in the pool you are checking only one here isn't it?